src/Cli/dotnet/commands/dotnet-sln/remove/Program.cs

```diff
@@ -114,21 +92,40 @@ private async Task RemoveProjectsAsync(string solutionFileFullPath, IEnumerable<
                 }
             }
 
-            Dictionary<SolutionFolderModel, SolutionItemModel[]> solutionItemsGroupedByParent = solution.SolutionItems
-                .Where(i => i.Parent != null)
-                .GroupBy(i => i.Parent)
-                .ToDictionary(g => g.Key, g => g.ToArray());
-
-            Dictionary<SolutionFolderModel, int> nonFolderDescendantsCount = new();
-            foreach (var item in solution.SolutionFolders)
+            for (int i = 0; i < solution.SolutionFolders.Count; i++)
             {
-                CountNonFolderDescendants(solution, item, solutionItemsGroupedByParent, nonFolderDescendantsCount);
-            }
+                var folder = solution.SolutionFolders[i];
+                int nonFolderDescendants = 0;
+                Stack<SolutionFolderModel> stack = new();
+                stack.Push(folder);
 
-            var emptyFolders = nonFolderDescendantsCount.Where(i => i.Value == 0).Select(i => i.Key);
-            foreach (var folder in emptyFolders)
-            {
-                solution.RemoveFolder(folder);
+                while (stack.Count > 0)
+                {
+                    var current = stack.Pop();
+
+                    nonFolderDescendants += current.Files?.Count ?? 0;
```

Comment:
I don't know if I understand this part properly, but this seems like it only goes one level down? Like if a SolutionFolder contains another folder that's also in SolutionItems, it'll look at that one, but if that folder contains a folder that isn't a SolutionItem, it ignores it? Why is that?